pkg/downloader/manager.go

```diff
@@ -220,42 +220,58 @@ func (m *Manager) downloadAll(deps []*chartutil.Dependency) error {
 	fmt.Fprintf(m.Out, "Saving %d charts\n", len(deps))
 	var saveError error
 	for _, dep := range deps {
+		var isSubChart bool
 		if strings.HasPrefix(dep.Repository, "file://") {
-			if m.Debug {
-				fmt.Fprintf(m.Out, "Archiving %s from repo %s\n", dep.Name, dep.Repository)
-			}
-			ver, err := tarFromLocalDir(m.ChartPath, dep.Name, dep.Repository, dep.Version)
+			//check if file repo path does belong to the current chart "/charts" folder
+			isSubChart, err = resolver.IsLocalSubChart(dep.Repository, m.ChartPath)
 			if err != nil {
 				saveError = err
-				break
-			}
-			dep.Version = ver
-			continue
+			} //else everythig is fine.
 		}
 
-		fmt.Fprintf(m.Out, "Downloading %s from repo %s\n", dep.Name, dep.Repository)
+		//Only download or package dependencies that are not sub chart in the "/charts" folder
+		if !isSubChart {
+			if strings.HasPrefix(dep.Repository, "file://") {
```

Comment:
same question as above; why are we checking if the repository has this prefix three times in the same function call? Seems like this needs to be optimized to perform the check only once.

Reply:
Here we are not a subchart of the '/charts' folder but we may have a `file://` dependency and we need to hande this deps in a different manner that remote dependencies.